packages/neon-core/__tests__/tx/transaction/Transaction.ts

```diff
@@ -1,31 +1,290 @@
-    expect(result.inputs.length).toBe(1);
-    expect(result.outputs.length).toBe(1);
-    expect(result.scripts.length).toBe(0);
+  test("addWitness", () => {
+    const tx1 = createTxforTestAddMethods();
+    tx1.addWitness(
+      new Witness({
+        invocationScript: "ab",
+        verificationScript:
+          "210317595a739cfe90ea90b6392814bcdebcd4c920cb149d0ac2d88676f1b0894fba68747476aa"
+      })
+    );
+    expect(tx1.scripts[0].invocationScript).toBe("ab");
+    expect(tx1.scripts[0].verificationScript).toBe(
+      "210317595a739cfe90ea90b6392814bcdebcd4c920cb149d0ac2d88676f1b0894fba68747476aa"
+    );
   });
 
-  test("deserialize signed tx", () => {
-    const result = Transaction.deserialize(
-      "800000022d9848a35942a8860b944d6c5f51c358bf2a8fbced8fd94975f38212cd448f7b0200b0c73210bc15bd74a25c07dcb7110f1d1be02c7818a5f9f8157af1ee5d7e18260100049b7cffdaa674beae0f930ebe6085af9093e5fe56b34a5c220ccdcf6efc336fc500e1f505000000003775292229eccdf904f16fff8e83e7cffdc0f0cee72d286979ee6cb1b7e65dfddfb2e384100b8d148e7758de42e4168b71792c6001000000000000003775292229eccdf904f16fff8e83e7cffdc0f0ce9b7cffdaa674beae0f930ebe6085af9093e5fe56b34a5c220ccdcf6efc336fc500c2eb0b0000000035b20010db73bf86371075ddfba4e6596f1ff35de72d286979ee6cb1b7e65dfddfb2e384100b8d148e7758de42e4168b71792c60e60300000000000035b20010db73bf86371075ddfba4e6596f1ff35d014140053539269d07c6d6a14f2f05c3676f97b803fe1cf6e10480878ca59b4713d58bf7f6864e817e760133633104b070657d36f662784493a2c8df63777cc2037c7d2321031d8e1630ce640966967bc6d95223d21f44304133003140c3b52004dc981349c9ac"
+  test("sign", () => {
+    const tx1 = createTxforTestAddMethods();
+    const account = new Account(
+      "9600debdb033bae62179baadb439c65088a450d5eecff782f641778fab23e21d"
+    );
+    tx1.scripts = [];
+    tx1.sign(account);
+    expect(tx1.scripts[0].verificationScript).toBe(
+      "210317595a739cfe90ea90b6392814bcdebcd4c920cb149d0ac2d88676f1b0894fba68747476aa"
+    );
+    expect(tx1.scripts[0].invocationScript).toBe(
+      "4073b9a44d007966c19e3010202aeed6dd9a158ab2b49d42eb14a3ab43509112674117a3e4cb0c9eadb45e93db621a3d0d6aa1a66086dc55b08a8e28de740e7dd7"
     );
+  });
+
+  test("addIntents", () => {
+    const tx1 = createTxforTestAddMethods();
+    tx1.addIntents({ scriptHash: "NEO", operation: "transfer", args: [10] });
+    expect(tx1.intents.length).toBe(1);
+    expect(tx1.script).toBe("005a51c1087472616e736665726845c49284");
+  });
+});
+
+describe("Fee Related", () => {
+  function createTxforTestFeeMethods(): Transaction {
+    return new Transaction({
+      version: 0,
+      nonce: 123,
+      sender: "39e9c91012be63a58504e52b7318c1274554ae3d",
+      systemFee: 12,
+      networkFee: 0,
+      validUntilBlock: 1000,
+      attributes: [],
+      scripts: [],
+      intents: [{ scriptHash: "NEO", operation: "transfer", args: [10] }]
+    });
+  }
+
+  describe("calculateNetworkFee", () => {
+    test("true as param to assign networkFee", () => {

```

Comment:
Why do we want to have this boolean? If we have cases to calculate the fees without affecting the transaction, then the method should be extracted out as a static method and the transaction class calling that method.

Reply:
Removed this boolean param